crypto/jwx.go

```diff
@@ -338,6 +434,35 @@ func SignatureAlgorithm(key crypto.PublicKey) (jwa.SignatureAlgorithm, error) {
 	}
 }
 
+func EncryptionAlgorithm(key crypto.PublicKey) (jwa.KeyEncryptionAlgorithm, error) {
+	if key == nil {
+		return "", errors.New("no key provided")
+	}
+
+	var ptr interface{}
+	switch v := key.(type) {
+	case crypto.PublicKey:
+		ptr = &v
+	case rsa.PublicKey:
+		ptr = &v
+	case ecdsa.PublicKey:
+		ptr = &v
+	default:
+		ptr = v
+	}
+
+	switch ptr.(type) {
+	case *crypto.PublicKey:
+		return defaultEcEncryptionAlgorithm, nil
+	case *rsa.PublicKey:
+		return defaultRsaEncryptionAlgorithm, nil
+	case *ecdsa.PublicKey:
+		return defaultEcEncryptionAlgorithm, nil
+	default:
+		return "", fmt.Errorf(`could not determine signature algorithm for key type '%T'`, key)
```

Comment:
No need for backtiks here
```suggestion
		return "", fmt.Errorf("could not determine signature algorithm for key type '%T'", key)
```